bfsos/iterator.go

```diff
@@ -0,0 +1,61 @@
+package bfsos
+
+import (
+	"sync"
+)
+
+// iterator implements an iterator over file path list.
+type iterator struct {
+	mutex sync.Mutex
```

Comment:
remove mutex. iterators are not supposed to be thread-safe, use same method as inmem one